app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt

```diff
@@ -66,6 +67,7 @@ import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 import timber.log.Timber
 
+@SuppressLint("NoLifecycleObserver") // we don't use DI here
```

Comment:
I don't see where the `SettingsViewModel` observers the activity lifecycle. Maybe we need to fix that?

Reply:
mmm...you are right...not sure what's this doing here. I see `onStop` cancels `deviceShieldStatePollingJob`, which is always null. Gonna remove it, but please review it because probably related to AppTp old feature?